test/new_relic/agent/method_tracer_params_test.rb

```diff
@@ -0,0 +1,178 @@
+    super
+  end
+
+  class UntracedMethods
+    def expect_deprecation_warnings?
+      RUBY_VERSION >= "2.7.0" && RUBY_VERSION < "3.0.0"
+    end
+
+    def no_args
+      {foo: {bar: "foobar"}}
+    end
+    def last_arg_expects_a_hash foo, bar = {}
+      {foo => bar}
+    end
+    def last_arg_is_a_keyword foo, bar:
+      {foo => bar}
+    end
+    def all_args_are_keywords(foo: '', bar: '')
+      {foo => bar}
+    end
+    def wildcard_args *args
+      { args[0] => args[1] }
+    end
+    def args_and_kwargs *args, **kwargs
+      {args[0] => kwargs}
+    end
+
+    def modifies_hash
+      (1..3).each_with_object({}) do |i, hash|
+        process(i, hash)
+      end
+    end
+  
+    def process(i, hash)
+      hash[i] = i * 3
+    end
+  end
+
+  class TracedMethods < UntracedMethods
+    include NewRelic::Agent::MethodTracer
+  
+    add_method_tracer :no_args
+    add_method_tracer :last_arg_expects_a_hash
+    add_method_tracer :last_arg_is_a_keyword
+    add_method_tracer :all_args_are_keywords
+    add_method_tracer :wildcard_args
+    add_method_tracer :args_and_kwargs
+    add_method_tracer :process
+  end
+  
+  class TracedMetricMethods < UntracedMethods
+    add_method_tracer :no_args, METRIC
+    add_method_tracer :last_arg_expects_a_hash, METRIC
+    add_method_tracer :last_arg_is_a_keyword, METRIC
+    add_method_tracer :all_args_are_keywords, METRIC
+    add_method_tracer :wildcard_args, METRIC
+    add_method_tracer :args_and_kwargs, METRIC
+    add_method_tracer :process, METRIC
+  end
+   
+  class TracedMetricMethodsUnscoped < UntracedMethods
+    add_method_tracer :no_args, METRIC, push_scope: false
+    add_method_tracer :last_arg_expects_a_hash, METRIC, push_scope: false
+    add_method_tracer :last_arg_is_a_keyword, METRIC, push_scope: false
+    add_method_tracer :all_args_are_keywords, METRIC, push_scope: false
+    add_method_tracer :wildcard_args, METRIC, push_scope: false
+    add_method_tracer :args_and_kwargs, METRIC, push_scope: false
+    add_method_tracer :process, METRIC, push_scope: false
+  end
+  
+  def refute_deprecation_warning
+    in_transaction do
+      _out, err = capture_io { yield }
+      refute_match KEYWORD_DEPRECATED_WARNING, err
+      return err
+    end
+  end
+
+  def assert_deprecation_warning
+    in_transaction do
+      _out, err = capture_io { yield }
+      assert_match KEYWORD_DEPRECATED_WARNING, err
+      return err
+    end
+  end
+
+  def silence_expected_warnings
+    capture_io { yield }
+  end
+
+  def assert_common_tracing_behavior traced_class
+    assert_expected_results traced_class
+    refute_deprecation_warnings traced_class
+    call_expecting_warning_after_ruby_26 traced_class
+  end
+
+  [ ["untraced_methods", UntracedMethods],
+    ["traced_methods", TracedMethods],
+    ["traced_metric_methods", TracedMetricMethods],
+    ["traced_metric_methods_unscoped", TracedMetricMethodsUnscoped],
+  ].each do |traced_class_name, traced_class, metrics|
```

Comment:
what is `metrics` here? I'm not seeing it used below, and it seems like each will only have the first 2 things

Reply:
need to remove `metrics`. no longer using it.